### ISSUE TITLE
about:killAllOnCompletedTaskNumber has been obsoleted

### ISSUE THREAD

**Issue by boozyguo** [Microsoft/pai#1135]
Hi, 
I am trying to submit a job by the webportal.  
The json file is cifar10.json from "Quick start: how to write and submit a CIFAR-10 job".
However, I got a error: "killAllOnCompletedTaskNumber has been obsoleted, please use minFailedTaskCount and "minSucceededTaskCount instead." 
I wonder how can fix the problem? 
Thank you!

**Comment by Gerhut** [comment]
`killAllOnCompletedTaskNumber` has been removed from the current master branch's [Quick Start section] (https://github.com/Microsoft/pai/tree/master/examples#quickstart), feel free to update yours and give us more feedback.

Ping @YanjieGao for a possible doc issue.

**Comment by fanyangCS** [comment]
If you deploy the latest bits from master branch, killAllOnCompletedTaskNumber has been obsoleted. If you use our stable release 0.6.y, etc. It has not yet. We suggest you deployment our stable release for the moment.

**Comment by YanjieGao** [comment]
current branch doc and example doesn't show killAllOnCompletedTaskNumber to user, i guess @boozyguo  write killAllOnCompletedTaskNumber  directly in job json file ?

**Comment by boozyguo** [comment]
@YanjieGao , thank you for help. I use the json file from  "Quick start: how to write and submit a CIFAR-10 job".

the  json file content is below:

{
  "jobName": "tensorflow-cifar10",
  "image": "openpai/pai.example.tensorflow",
  "dataDir": "/tmp/data",
  "outputDir": "/tmp/output",
  "taskRoles": [
    {
      "name": "cifar_train",
      "taskNumber": 1,
      "cpuNumber": 8,
      "memoryMB": 32768,
      "gpuNumber": 1,
      "command": "git clone https://github.com/tensorflow/models && cd models/research/slim && python download_and_convert_data.py --dataset_name=cifar10 --dataset_dir=$PAI_DATA_DIR && python train_image_classifier.py --batch_size=64 --model_name=inception_v3 --dataset_name=cifar10 --dataset_split_name=train --dataset_dir=$PAI_DATA_DIR --train_dir=$PAI_OUTPUT_DIR"
    }
  ]
}

There is no killAllOnCompletedTaskNumber  in json file.  
However, the "submit job" webportal  has the killAllOnCompletedTaskNumber  filed.

**Comment by YanjieGao** [comment]
@boozyguo  you could refer fan's answer. use the stable release of pai

**Comment by YanjieGao** [comment]
Hi @hao1939 , dev-box docker is use master branch for deployment, this may lead to customer use unstable version problem. Is it better each release use the specific release version of PAI?   
![image](https://user-images.githubusercontent.com/5576848/44566584-ed248680-a7a0-11e8-9efa-f01a0d655a14.png

**Comment by fanyangCS** [comment]
@boozyguo please go to https://github.com/Microsoft/pai/releases to checkout a latest release.

**Comment by boozyguo** [comment]
thank you. @fanyangCS @YanjieGao 

So,  I should download PAI v0.6.1 in dev-box, then  prepare quick-start.yaml and run paictl.py?

**Comment by hao1939** [comment]
Hi @boozyguo ,

You should match the image tags and code branch.
In case you are using currently latest release v0.6.1：
1. In dev-box, make sure checkout the code with tag v0.6.1
2. In config file `services-configuration.yaml`, set the `docker-tag: v0.6.1`

If you want to update the config, recommend fellow steps:
1. In dev-box, checkout the tag v0.6.1
2. generate the config (make sure generate in another directory, it won't override the previously config)
3. In config file `services-configuration.yaml`, set the `docker-tag: v0.6.1`
4. stop old installation, using `paictl.py service stop ...`
5. start with new config, using `paictl.py service start ...`

That's all.

**Comment by hao1939** [comment]
Hi @boozyguo ,

You should using `python paictl.py cluster generate-configuration ...`.
There are break changes after v0.6.1.
So please refer to the doc under the branch pai-0.6.y: https://github.com/Microsoft/pai/blob/pai-0.6.y/pai-management/doc/single-box-deployment.md

**Comment by boozyguo** [comment]
thank you @hao1939 
I have try :
https://github.com/Microsoft/pai/blob/pai-0.6.y/pai-management/doc/single-box-deployment.md
https://github.com/Microsoft/pai/blob/pai-0.6.y/pai-management/doc/cluster-bootup.md


however, I got another error, deploy failed.   
.........
2018-08-24 13:15:35,871 [ERROR] - k8sPaiLibrary.maintainlib.common : There will be a delay after installing, please wait.
Segmentation fault
2018-08-24 13:15:35,874 [ERROR] - k8sPaiLibrary.maintainlib.common : There will be a delay after installing, please wait.
Segmentation fault
2018-08-24 13:15:35,878 [ERROR] - k8sPaiLibrary.maintainlib.common : There will be a delay after installing, please wait.
Segmentation fault
2018-08-24 13:15:35,880 [ERROR] - k8sPaiLibrary.maintainlib.common : There will be a delay after installing, please wait.
Segmentation fault
2018-08-24 13:15:35,884 [ERROR] - k8sPaiLibrary.maintainlib.common : There will be a delay after installing, please wait.
Segmentation fault
..............

**Comment by hao1939** [comment]
Hi @boozyguo ,

The log is some kind of misleading, actually it's waiting for the pod to be ready.
Could you check the k8s dashboard in the same time?
It's on http://your_master_ip:9090.

**Comment by boozyguo** [comment]
Hello, @hao1939 
I have try other version of PAI, but there always are some network problem(can not pull images from gcr.io or docker.io).  
So, I choose the latest version. At "submit job" webpage, I found a label named "Properties". When click it, there are some items including "killAllOnCompletedTaskNumber".  After uncheck the "killAllOnCompletedTaskNumber" item, the job can submitted succed.

I consider there is a bug between "webportal" and "api-server" or another parts of PAI. They have a different version. 

Thank you for help.

**Comment by hao1939** [comment]
Hi @boozyguo ,

You are right, we fixed the network problem on the release v0.6.1, because in some region gcr.io is not reachable.

'api-server' and 'webportal' do have different version.
And you can check the images tag on k8s dashboard.
In you case, it should be `docker.io/openpai/rest-server:latest`.
![image](https://user-images.githubusercontent.com/1547343/44829635-51879000-ac50-11e8-8c4a-ce74d2a3f18f.png)

**Comment by hao1939** [comment]
Hi @boozyguo ,

I will close it for now, if you have more questions, feel free to reopen it.

**Comment by boozyguo** [comment]
hello, @hao1939  thank you for your help. I got the method to deal with the problem.
